crash on terraform apply

Here's the dump I'm getting: 

```
terraform apply -var "secret_key=$AWS_SECRET_ACCESS_KEY" -var "access_key=$AWS_ACCESS_KEY_ID"
aws_instance.nfconv_worker: Refreshing state... (ID: )
aws_security_group.nfconv-management: Refreshing state... (ID: sg-11b88474)
Error refreshing state: 2 error(s) occurred:

* 1 error(s) occurred:

* unexpected EOF
* 1 error(s) occurred:

* connection is shut down
panic: assignment to entry in nil map
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 31 [running]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.migrateStateV0toV1(0xc208065c20, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_instance_migrate.go:36 +0x20f
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsInstanceMigrateState(0x0, 0xc208065c20, 0x42d600, 0xc2080fdac0, 0x8036d00, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_instance_migrate.go:18 +0x123
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc20810f740, 0xc208065c20, 0x42d600, 0xc2080fdac0, 0xc2080f3ca0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/helper/schema/resource.go:193 +0x295
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc2081091a0, 0xc2080dd840, 0xc208065c20, 0x42e920, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/helper/schema/provider.go:179 +0x1c7
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Refresh(0xc2080e8680, 0xc2080f16a0, 0xc2080f1c90, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/resource_provider.go:341 +0x73
2015/03/25 17:59:19 terraform-provider-aws: reflect.Value.call(0x4f9940, 0x5ccaa0, 0x13, 0x60dc90, 0x4, 0xc2080d5f28, 0x3, 0x3, 0x0, 0x0, ...)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/reflect/value.go:419 +0x10e5
2015/03/25 17:59:19 terraform-provider-aws: reflect.Value.Call(0x4f9940, 0x5ccaa0, 0x13, 0xc2080d5f28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/reflect/value.go:296 +0xbc
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*service).call(0xc20810fd40, 0xc20810fd00, 0xc208111c50, 0xc208075180, 0xc2080e86a0, 0x42e8c0, 0xc2080f16a0, 0x16, 0x42e920, 0xc2080f1c90, ...)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:382 +0x1f7
2015/03/25 17:59:19 terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:476 +0x44a
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 1 [IO wait]:
2015/03/25 17:59:19 terraform-provider-aws: net.(*pollDesc).Wait(0xc208010840, 0x72, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
2015/03/25 17:59:19 terraform-provider-aws: net.(*pollDesc).WaitRead(0xc208010840, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:89 +0x43
2015/03/25 17:59:19 terraform-provider-aws: net.(*netFD).accept(0xc2080107e0, 0x0, 0xa39d48, 0xc20800b508)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:419 +0x40b
2015/03/25 17:59:19 terraform-provider-aws: net.(*UnixListener).AcceptUnix(0xc20801ed20, 0x36820, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/unixsock_posix.go:282 +0x56
2015/03/25 17:59:19 terraform-provider-aws: net.(*UnixListener).Accept(0xc20801ed20, 0x0, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/unixsock_posix.go:293 +0x4c
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*Server).Accept(0xc20800b400, 0xa3a5c0, 0xc20801ed20)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:33 +0x37
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/plugin.Serve(0xc20800b3e0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/plugin/server.go:88 +0x7e8
2015/03/25 17:59:19 terraform-provider-aws: main.main()
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/builtin/bins/provider-aws/main.go:11 +0x6d
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 5 [syscall]:
2015/03/25 17:59:19 terraform-provider-aws: os/signal.loop()
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/os/signal/signal_unix.go:21 +0x1f
2015/03/25 17:59:19 terraform-provider-aws: created by os/signal.init·1
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/os/signal/signal_unix.go:27 +0x35
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 7 [chan receive]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/plugin.func·007()
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/plugin/server.go:79 +0x69
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/terraform/plugin.Serve
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/plugin/server.go:85 +0x7c4
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 8 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Stream).Read(0xc20802e780, 0xc2080df000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).fill(0xc208052660)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:97 +0x1ce
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).Read(0xc208052660, 0xc20800b590, 0x1, 0x9, 0x2592, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:174 +0x26c
2015/03/25 17:59:19 terraform-provider-aws: io.ReadAtLeast(0xa3b718, 0xc208052660, 0xc20800b590, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:298 +0xf1
2015/03/25 17:59:19 terraform-provider-aws: io.ReadFull(0xa3b718, 0xc208052660, 0xc20800b590, 0x1, 0x9, 0x2c, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:316 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.decodeUintReader(0xa3b718, 0xc208052660, 0xc20800b590, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decode.go:121 +0x99
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).recvMessage(0xc208074700, 0xc2080017a0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:76 +0x55
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).decodeTypeSequence(0xc208074700, 0xc20801f400, 0x16)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:140 +0x47
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).DecodeValue(0xc208074700, 0x434620, 0xc20801f400, 0x16, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:208 +0x192
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).Decode(0xc208074700, 0x434620, 0xc20801f400, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:185 +0x297
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc208064300, 0xc20801f400, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:402 +0x5e
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequestHeader(0xc208035fc0, 0xa3b878, 0xc208064300, 0x0, 0x0, 0xc20801f400, 0xc20800b500, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:575 +0x97
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequest(0xc208035fc0, 0xa3b878, 0xc208064300, 0xc208074600, 0xc20801ee80, 0x450b80, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:542 +0xa2
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeCodec(0xc208035fc0, 0xa3b878, 0xc208064300)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:461 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeConn(0xc208035fc0, 0xa3b7a0, 0xc20802e780)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:453 +0x208
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*Server).ServeConn(0xc20800b400, 0xa3b6b8, 0xc208038098)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:76 +0x530
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*Server).Accept
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:39 +0x173
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 9 [IO wait]:
2015/03/25 17:59:19 terraform-provider-aws: net.(*pollDesc).Wait(0xc208010920, 0x72, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:84 +0x47
2015/03/25 17:59:19 terraform-provider-aws: net.(*pollDesc).WaitRead(0xc208010920, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_poll_runtime.go:89 +0x43
2015/03/25 17:59:19 terraform-provider-aws: net.(*netFD).Read(0xc2080108c0, 0xc20800f000, 0x1000, 0x1000, 0x0, 0xa39d48, 0xc2080f1650)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/fd_unix.go:242 +0x40f
2015/03/25 17:59:19 terraform-provider-aws: net.(*conn).Read(0xc208038098, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/net.go:121 +0xdc
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).fill(0xc208052300)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:97 +0x1ce
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).Read(0xc208052300, 0xc20800b520, 0xc, 0xc, 0xc2080f1640, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:174 +0x26c
2015/03/25 17:59:19 terraform-provider-aws: io.ReadAtLeast(0xa3b718, 0xc208052300, 0xc20800b520, 0xc, 0xc, 0xc, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:298 +0xf1
2015/03/25 17:59:19 terraform-provider-aws: io.ReadFull(0xa3b718, 0xc208052300, 0xc20800b520, 0xc, 0xc, 0xc, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:316 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).recv(0xc208044d20)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:340 +0xcb
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/yamux.newSession
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:93 +0x3f0
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 10 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).send(0xc208044d20)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:298 +0x554
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/yamux.newSession
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:94 +0x40a
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 11 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).keepalive(0xc208044d20)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:254 +0x106
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/yamux.newSession
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:96 +0x435
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 13 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Session).AcceptStream(0xc208044d20, 0xc200000003, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/session.go:162 +0x173
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*muxBroker).Run(0xc20801ee40)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/mux_broker.go:107 +0x37
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*Server).ServeConn
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:65 +0x388
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 15 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Stream).Read(0xc20802e140, 0xc2080af000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).fill(0xc208052cc0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:97 +0x1ce
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).Read(0xc208052cc0, 0xc20800b175, 0x1, 0x9, 0xc2080d1858, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:174 +0x26c
2015/03/25 17:59:19 terraform-provider-aws: io.ReadAtLeast(0xa3b718, 0xc208052cc0, 0xc20800b175, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:298 +0xf1
2015/03/25 17:59:19 terraform-provider-aws: io.ReadFull(0xa3b718, 0xc208052cc0, 0xc20800b175, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:316 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.decodeUintReader(0xa3b718, 0xc208052cc0, 0xc20800b175, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decode.go:121 +0x99
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).recvMessage(0xc208074780, 0xc2080885a0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:76 +0x55
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).decodeTypeSequence(0xc208074780, 0xc20801ff00, 0x16)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:140 +0x47
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).DecodeValue(0xc208074780, 0x434620, 0xc20801ff60, 0x16, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:208 +0x192
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).Decode(0xc208074780, 0x434620, 0xc20801ff60, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:185 +0x297
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080655f0, 0xc20801ff60, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:402 +0x5e
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequestHeader(0xc2080dd280, 0xa3b878, 0xc2080655f0, 0x0, 0x0, 0xc20801ff60, 0xc20800b100, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:575 +0x97
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequest(0xc2080dd280, 0xa3b878, 0xc2080655f0, 0xc208074280, 0xc20801f580, 0x42e800, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:542 +0xa2
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeCodec(0xc2080dd280, 0xa3b878, 0xc2080655f0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:461 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeConn(0xc2080dd280, 0xa3b7a0, 0xc20802e140)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:453 +0x208
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.serve(0xa3b7a0, 0xc20802e140, 0x674f10, 0x10, 0x5cc940, 0xc20801f560)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.func·001()
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 20 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Stream).Read(0xc20802e500, 0xc208105000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).fill(0xc208053800)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:97 +0x1ce
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).Read(0xc208053800, 0xc2080f0245, 0x1, 0x9, 0xc2080d3858, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:174 +0x26c
2015/03/25 17:59:19 terraform-provider-aws: io.ReadAtLeast(0xa3b718, 0xc208053800, 0xc2080f0245, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:298 +0xf1
2015/03/25 17:59:19 terraform-provider-aws: io.ReadFull(0xa3b718, 0xc208053800, 0xc2080f0245, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:316 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.decodeUintReader(0xa3b718, 0xc208053800, 0xc2080f0245, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decode.go:121 +0x99
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).recvMessage(0xc208074e00, 0xc208000d80)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:76 +0x55
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).decodeTypeSequence(0xc208074e00, 0xc2080ea500, 0x16)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:140 +0x47
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).DecodeValue(0xc208074e00, 0x434620, 0xc2080ea540, 0x16, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:208 +0x192
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).Decode(0xc208074e00, 0x434620, 0xc2080ea540, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:185 +0x297
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080e53e0, 0xc2080ea540, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:402 +0x5e
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequestHeader(0xc2080fc440, 0xa3b878, 0xc2080e53e0, 0x0, 0x0, 0xc2080ea540, 0xc2080f0200, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:575 +0x97
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequest(0xc2080fc440, 0xa3b878, 0xc2080e53e0, 0xc208074d80, 0xc2080eafe0, 0x42e9e0, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:542 +0xa2
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeCodec(0xc2080fc440, 0xa3b878, 0xc2080e53e0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:461 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeConn(0xc2080fc440, 0xa3b7a0, 0xc20802e500)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:453 +0x208
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.serve(0xa3b7a0, 0xc20802e500, 0x674f10, 0x10, 0x5cc940, 0xc2080ea520)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.func·001()
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/25 17:59:19 terraform-provider-aws:
2015/03/25 17:59:19 terraform-provider-aws: goroutine 27 [select]:
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/yamux.(*Stream).Read(0xc20802e3c0, 0xc208122000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).fill(0xc208052960)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:97 +0x1ce
2015/03/25 17:59:19 terraform-provider-aws: bufio.(*Reader).Read(0xc208052960, 0xc208111bf5, 0x1, 0x9, 0xc208119858, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/bufio/bufio.go:174 +0x26c
2015/03/25 17:59:19 terraform-provider-aws: io.ReadAtLeast(0xa3b718, 0xc208052960, 0xc208111bf5, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:298 +0xf1
2015/03/25 17:59:19 terraform-provider-aws: io.ReadFull(0xa3b718, 0xc208052960, 0xc208111bf5, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/io/io.go:316 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.decodeUintReader(0xa3b718, 0xc208052960, 0xc208111bf5, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decode.go:121 +0x99
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).recvMessage(0xc208075380, 0xc208088b40)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:76 +0x55
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).decodeTypeSequence(0xc208075380, 0xc2080e9100, 0x16)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:140 +0x47
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).DecodeValue(0xc208075380, 0x434620, 0xc2080e9140, 0x16, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:208 +0x192
2015/03/25 17:59:19 terraform-provider-aws: encoding/gob.(*Decoder).Decode(0xc208075380, 0x434620, 0xc2080e9140, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/encoding/gob/decoder.go:185 +0x297
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2081094d0, 0xc2080e9140, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:402 +0x5e
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequestHeader(0xc20810fd00, 0xa3b878, 0xc2081094d0, 0x0, 0x0, 0xc2080e9140, 0xc208111b00, 0x0, 0x0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:575 +0x97
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).readRequest(0xc20810fd00, 0xa3b878, 0xc2081094d0, 0xc208075180, 0xc2080e86a0, 0x42e8c0, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:542 +0xa2
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeCodec(0xc20810fd00, 0xa3b878, 0xc2081094d0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:461 +0x6d
2015/03/25 17:59:19 terraform-provider-aws: net/rpc.(*Server).ServeConn(0xc20810fd00, 0xa3b7a0, 0xc20802e3c0)
2015/03/25 17:59:19 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:453 +0x208
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.serve(0xa3b7a0, 0xc20802e3c0, 0x674f10, 0x10, 0x5cc940, 0xc2080e8680)
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/25 17:59:19 terraform-provider-aws: github.com/hashicorp/terraform/rpc.func·001()
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/25 17:59:19 terraform-provider-aws: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/25 17:59:19 terraform-provider-aws: 	/Users/spencer/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalRefresh, err: unexpected EOF
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalSequence, err: unexpected EOF
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalOpFilter, err: unexpected EOF
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalSequence, err: unexpected EOF
2015/03/25 17:59:19 [DEBUG] /Users/spencer/go/bin/terraform-provider-aws: plugin process exited
2015/03/25 17:59:19 [DEBUG] vertex root.aws_security_group.nfconv-management: walking
2015/03/25 17:59:19 [DEBUG] vertex root.aws_security_group.nfconv-management: evaluating
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalSequence
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalInterpolate
2015/03/25 17:59:19 [DEBUG] root: eval: terraform.EvalNoop
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalCountFixZeroOneBoundary
2015/03/25 17:59:19 [DEBUG] vertex root.aws_security_group.nfconv-management: expanding/walking dynamic subgraph
2015/03/25 17:59:19 [TRACE] Graph after step *terraform.ResourceCountTransformer:

aws_security_group.nfconv-management
2015/03/25 17:59:19 [TRACE] Graph after step *terraform.RootTransformer:

aws_security_group.nfconv-management
2015/03/25 17:59:19 [DEBUG] vertex root.aws_security_group.nfconv-management: walking
2015/03/25 17:59:19 [DEBUG] vertex root.aws_security_group.nfconv-management: evaluating
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalSequence
2015/03/25 17:59:19 [DEBUG] root: eval: terraform.EvalNoop
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalInstanceInfo
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalOpFilter
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalSequence
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalGetProvider
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalReadState
2015/03/25 17:59:19 [DEBUG] root: eval: *terraform.EvalRefresh
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalRefresh, err: connection is shut down
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalSequence, err: connection is shut down
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalOpFilter, err: connection is shut down
2015/03/25 17:59:19 [ERROR] root: eval: *terraform.EvalSequence, err: connection is shut down
2015/03/25 17:59:19 [DEBUG] vertex root.root: walking
2015/03/25 17:59:19 waiting for all plugin processes to complete...
2015/03/25 17:59:19 [DEBUG] /Users/spencer/go/bin/terraform-provisioner-local-exec: plugin process exited
2015/03/25 17:59:19 [DEBUG] /Users/spencer/go/bin/terraform-provisioner-remote-exec: plugin process exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

This consistently happens. Retrying isn't helping.

My version:
```
terraform version
Terraform v0.4.0-dev (55d682482a072f783de33954bb396bbf556a882c)```

--- a/builtin/providers/aws/resource_aws_instance_migrate.go
+++ b/builtin/providers/aws/resource_aws_instance_migrate.go
@@ -24,7 +24,13 @@ func resourceAwsInstanceMigrateState(
 }
 
 func migrateStateV0toV1(is *terraform.InstanceState) (*terraform.InstanceState, error) {
+	if is.Empty() {
+		log.Println("[DEBUG] Empty InstanceState; nothing to migrate.")
+		return is, nil
+	}
+
 	log.Printf("[DEBUG] Attributes before migration: %#v", is.Attributes)
+
 	// Delete old count
 	delete(is.Attributes, "block_device.#")
 

--- a/builtin/providers/aws/resource_aws_instance_migrate_test.go
+++ b/builtin/providers/aws/resource_aws_instance_migrate_test.go
@@ -115,6 +115,7 @@ func TestAWSInstanceMigrateState(t *testing.T) {
 
 	for tn, tc := range cases {
 		is := &terraform.InstanceState{
+			ID:         "i-abc123",
 			Attributes: tc.Attributes,
 		}
 		is, err := resourceAwsInstanceMigrateState(
@@ -131,5 +132,28 @@ func TestAWSInstanceMigrateState(t *testing.T) {
 					tn, k, v, k, is.Attributes[k], is.Attributes)
 			}
 		}
+	}
+}
+
+func TestAWSInstanceMigrateState_empty(t *testing.T) {
+	var is *terraform.InstanceState
+	var meta interface{}
+
+	// should handle nil
+	is, err := resourceAwsInstanceMigrateState(0, is, meta)
+
+	if err != nil {
+		t.Fatalf("err: %#v", err)
+	}
+	if is != nil {
+		t.Fatalf("expected nil instancestate, got: %#v", is)
+	}
+
+	// should handle non-nil but empty
+	is = &terraform.InstanceState{}
+	is, err = resourceAwsInstanceMigrateState(0, is, meta)
+
+	if err != nil {
+		t.Fatalf("err: %#v", err)
 	}
 }

--- a/terraform/state.go
+++ b/terraform/state.go
@@ -885,6 +885,10 @@ func (i *InstanceState) deepcopy() *InstanceState {
 	return n
 }
 
+func (s *InstanceState) Empty() bool {
+	return s == nil || s.ID == ""
+}
+
 func (s *InstanceState) Equal(other *InstanceState) bool {
 	// Short circuit some nil checks
 	if s == nil || other == nil {

--- a/terraform/state_test.go
+++ b/terraform/state_test.go
@@ -366,6 +366,34 @@ func TestResourceStateTaint(t *testing.T) {
 	}
 }
 
+func TestInstanceStateEmpty(t *testing.T) {
+	cases := map[string]struct {
+		In     *InstanceState
+		Result bool
+	}{
+		"nil is empty": {
+			nil,
+			true,
+		},
+		"non-nil but without ID is empty": {
+			&InstanceState{},
+			true,
+		},
+		"with ID is not empty": {
+			&InstanceState{
+				ID: "i-abc123",
+			},
+			false,
+		},
+	}
+
+	for tn, tc := range cases {
+		if tc.In.Empty() != tc.Result {
+			t.Fatalf("%q expected %#v to be empty: %#v", tn, tc.In, tc.Result)
+		}
+	}
+}
+
 func TestInstanceStateEqual(t *testing.T) {
 	cases := []struct {
 		Result   bool

